src/prog_models/sim_result.py

```diff
@@ -65,14 +65,14 @@ def pop(self, index : int = -1) -> dict:
         self.times.pop(index)
         return self.data.pop(index)
     
-    def remove(self, index : int) -> None:
+    def remove(self, target : int) -> None:
```

Comment:
```suggestion
    def remove(self, target) -> None:
```
I just realized, for remove the target should actually be type dict (since it's removing by value. I think it's best to keep the hint off for now